s3/pkg/datastore/yig/config/enum.go

```diff
@@ -40,7 +40,8 @@ func ReadConfigs(dir string, funcConfigParse FuncConfigParse) error {
 		viper.SetConfigFile(path)
 		err = viper.ReadInConfig()
 		if err != nil {
-			return err
+			// skip the config file which is failed to parse and continue the next one.
+			return nil
```

Comment:
What if read config file failed? Anything will be used as default ?

Reply:
if we put some other file which is not toml file, viper ReadInConfig will return an error. In this case, we should skip this kind of file and continue to parse other files till we find our needed config files.

Reply:
I mean ... if there is no valid config files, then what will happen?

Reply:
the parse will fail, and we cannot get the configs needed by yig, so, yig will not create valid drivers and when request comes in, yig will return error when handle it.